doc/api/esm.md

```diff
@@ -687,12 +691,39 @@ provided via a `--experimental-loader ./loader-name.mjs` argument to Node.js.
 When hooks are used they apply to the entry point and all `import` calls. They
 won't apply to `require` calls; those still follow [CommonJS][] rules.
 
+Loaders follow the pattern of `--require`:
+
+```console
+node \
+  --experimental-loader unpkg \
+  --experimental-loader http-to-https \
+  --experimental-loader cache-buster
+```
+
+These are called in the following sequence: `cache-buster` calls
+`http-to-https` which calls `unpkg`. Or in terms of simple functions,
+`cacheBuster(httpToHttps(unpkg(input)))`.
+
 ### Hooks
 
-#### `resolve(specifier, context, defaultResolve)`
+Hooks are part of a chain, even if that chain consists of only one custom
+(user-provided) hook and the default hook, which is always present. Hook
+functions nest: each one must always return a plain object, and chaining happens
+as a result of each function calling `next()`, which is a reference to the
+subsequent loader’s hook.
+
+A hook that fails to return triggers an exception. A hook that returns without
```

Comment:
> A hook that fails to return triggers an exception.

What does this mean?  I can think of several interpretations: the hook throws a sync exception, it returns a rejected promise, it returns a promise that does not resolve after a certain timeout, it returns a non-promise, or it returns a promise that resolves to a non-object value?

Reply:
it literally throws and the process exists with a non-zero status

Reply:
Thanks, sorry, I asked my question incorrectly.  I meant, what does it mean for a hook to fail to return?  Like, what does a hook need to do to "fail to return?"

Reply:
The return value is not as described for each hook below (an object with the necessary properties).

Reply:
Is this any clearer?

```suggestion
A hook neglecting to return triggers an exception. A hook that returns without
```

Reply:
I don't think it's possible to "fail to return", I initially though you meant "awaits a never settling promise". I think we could say:
```suggestion
A hook that returns a nullish value triggers an exception. A hook that returns without
```

or even (since returning `undefined` technically fits the other case):
```suggestion
A hook that returns without
```

Reply:
Even then, "properly return" is not a well defined concept x) wdyt of the following?
```suggestion
A hook that returns a value that lacks a required property triggers an exception. A hook that returns without
```

Reply:
(I still think we could skip the first sentence, required properties being required seems obvious to me; it's not like we are saying that the second sentence is the only way of triggering a exception in a hook)

Reply:
sure, works for me. pushing